[SHK-11] DeliveryInfo 저장 / 조회 / 수정 / 삭제 기능 구현

BnDC opened the pull request:
### ⛏ 작업 사항
- 배송 정보 저장 기능 구현 및 테스트
- 배송 정보 조회 기능 구현 및 테스트
- 배송 정보 수정 기능 구현 및 테스트
- 배송 정보 삭제 기능 구현 및 테스트

### 📝 작업 요약
배송 정보 저장 / 조회 / 수정 / 삭제 기능 구현 및 테스트

### 💡 관련 이슈
- Resolved : [SHK-191], [SHK-192], [SHK-193], [SHK-194], [SHK-225], [SHK-226], [SHK-227], [SHK-228]


[SHK-191]: https://shoekream.atlassian.net/browse/SHK-191?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[SHK-192]: https://shoekream.atlassian.net/browse/SHK-192?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[SHK-193]: https://shoekream.atlassian.net/browse/SHK-193?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[SHK-194]: https://shoekream.atlassian.net/browse/SHK-194?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[SHK-225]: https://shoekream.atlassian.net/browse/SHK-225?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[SHK-226]: https://shoekream.atlassian.net/browse/SHK-226?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[SHK-227]: https://shoekream.atlassian.net/browse/SHK-227?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[SHK-228]: https://shoekream.atlassian.net/browse/SHK-228?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

BnDC committed:
Feat : [SHK-191] 배송 정보 조회 기능 구현

BnDC committed:
Test : [SHK-192] 배송 정보 조회 기능 테스트

BnDC committed:
Feat : [SHK-193] 배송 정보 저장 기능 구현

BnDC committed:
Test : [SHK-194] 배송 정보 저장 기능 테스트

BnDC committed:
Feat : [SHK-225] 배송 정보 수정 기능 구현

BnDC committed:
Test : [SHK-226] 배송 정보 수정 기능 테스트

BnDC committed:
Feat : [SHK-227] 배송 정보 삭제 기능 구현

BnDC committed:
Test : [SHK-228] 배송 정보 삭제 기능 테스트

BnDC committed:
Test : 사용자 정보 수정 성공 테스트 수정
- 컨벤션에 맞게 줄맞춤

boompatron commented on src/main/java/com/prgrms/kream/domain/member/dto/request/DeliveryInfoDeleteRequest.java:
```diff
@@ -0,0 +1,11 @@
+
+public record DeliveryInfoDeleteRequest(
+		@NotNull
+		@Positive(message = "deliveryInfoId는 자연수여야 합니다.")
```
아이디가 음수로 넘어올 수도 있다는 생각을 안해봤는데, 사용자를 믿지 말라고 했으니...
저도 나중에 리팩토링 할 때 검증 조건에 추가해보겠습니다!

boompatron commented on src/main/java/com/prgrms/kream/domain/member/dto/request/DeliveryInfoRegisterRequest.java:
```diff
@@ -0,0 +1,29 @@
+		@Length(max = 20, message = "이름은 20자까지만 허용됩니다.")
+		String name,
+
+		@NotBlank(message = "이름은 빈값일 수 없습니다.")
```
위의 검증 메세지랑 중복된 것 같아요!

BnDC replied:
오 감사합니다 복사하는 과정에서 수정을 못한 거 같아요 수정하겠습니다.

boompatron commented on src/main/java/com/prgrms/kream/domain/member/dto/request/DeliveryInfoRegisterRequest.java:
```diff
@@ -0,0 +1,29 @@
+		@Length(max = 30, message = "상세 주소는 30자 이하여야 합니다.")
+		String detail,
+
+		@Positive(message = "memberId는 음수가 될 수 없습니다.")
```
memberid는 양수여야하는 동시에 빈 값이 들어오면 안되는 것으로 알고 있는데 
@Positive를 사용하면 둘 다 동시에 잡을 수 있나요??

BnDC replied:
memberId에 `@NotNull`이 누락된거 같습니다. 말씀해주신 부분과 함께 request dto에 bean validation을 전반적으로 점검하겠습니다.

boompatron reviewed (approved):
구현하느라 수고하셨습니다!! 몇 가지 질문 남겼어요!

BnDC committed:
Refactor : 누락된 bean validation 추가

- DeliveryInfoRegisterRequest에 누락된 validation 추가
- DeliveryInfoUpdateRequest에 누락된 validation 추가
- DeliveryInfoDeleteRequest에 누락된 validation 추가

Kim-Changgyu commented on src/main/java/com/prgrms/kream/domain/member/controller/MemberController.java:
```diff
@@ -90,4 +100,39 @@ public ApiResponse<MemberUpdateResponse> update(
+
+	@GetMapping("/{id}/delivery-infos")
+	@ResponseStatus(OK)
+	public ApiResponse<Page<DeliveryInfoGetResponse>> getDeliveryInfoPage(
```
응답은 page를 이용하셨군요!
혹시 사용자 당 Delivery Info의 개수가 많아질 수 있기 때문인가요?

BnDC replied:
네 맞습니다.

Kim-Changgyu reviewed (approved):
고생하셨습니다 ! 👍 
특별히 코멘트 드릴게 없습니다 :)
